File: leetcode/solutions/0345.ReverseVowelsOfAString/reverseVowels_test.go
package solution

import (
	"testing"

	"github.com/google/go-cmp/cmp"
)

func Test_0345_ReverseVowelsOfAString(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  string
		out string
	}{
		{"hello", "holle"},
		{"leetcode", "leotcede"},
		{"aA", "Aa"},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := reverseVowels(tt.in)

			if !cmp.Equal(got, tt.out) {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0345.ReverseVowelsOfAString/reverseVowels_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import reverseVowels from "./reverseVowels.ts";

Deno.test("0345. Reverse Vowels of a String", () => {
  type test = [string, string];
  const tests: test[] = [
    ["hello", "holle"],
    ["leetcode", "leotcede"],
    ["aA", "Aa"],
  ];

  for (const [got, want] of tests) {
    assertEquals(reverseVowels(got), want);
  }
});
